fdbserver/MasterProxyServer.actor.cpp

```diff
@@ -1431,7 +1401,7 @@ ACTOR Future<GetReadVersionReply> getLiveCommittedVersion(ProxyCommitData* commi
 			rep = v;
 		}
 	}
-	rep.recentRequests = commitData->stats.getRecentRequests();
+	rep.processBusyTime = 1e6 * (g_network->isSimulated() ? deterministicRandom()->random01() : g_network->networkInfo.metrics.lastZeroBusy);
```

Comment:
If we are susceptible to being de-scheduled like that, we may have quite a few issues with these types of busyness metrics in general, and potentially also to our latencies.

Ignoring the complexities of what one of these environments will do, we have some experience with this type of phenomenon by running more processes than we have cores and using all of the machine CPU resources. In these cases, we observe long slow tasks and inaccurate process busyness metrics (in addition to a variety of other problems which may be more specific to this case). I'm not sure it's really clear what the result will be for a process that is denied CPU -- it will probably record extra time at the priority it was running when it got de-scheduled, so that could mean a busyness estimate that is too high or too low depending on when it happens.

There are some guardrails in this algorithm to prevent it from having a huge disparity in how much is allocated to each proxy, but we should make sure that it at least doesn't go crazy when we see machine CPU saturation. We should also determine what we expect to see in something like our recommended k8s configuration.

Reply:
Considering de-scheduling and resource contention at the process will be tough and expensive, because process shall not know when it is descheduled. 

A simple solution will be making it clear in our deployment or operation guide that FDB should be deployed in a dedicated CPU and preferable dedicated machine (even on k8s). 

We may want to profile how much CPU resource each process gets from OS or hardware, so that we know if k8s breaks any resource contract.